syncr_backend/external_interface/dht_util.py

```diff
@@ -0,0 +1,84 @@
+import asyncio
+from typing import Any
+from typing import List
+from typing import Tuple
+
+from kademlia.network import Server  # type: ignore
+from kademlia.storage import ForgetfulStorage  # type: ignore
+
+from syncr_backend.util import crypto_util
+from syncr_backend.util.log_util import get_logger
+
+
+logger = get_logger(__name__)
+
+_node_instance = None
+
+
+def get_dht() -> Server:
+    """
+    returns the node_instance of the dht
+    if no node instance has been created it
+    connects to the distributed hash table
+    if no bootstrap ip port pair list is given, it starts a new dht
+    :param bootstrap_ip_port_pair_list:
+    list of ip port tuples to connect to the dht
+    :param listen_port: port to listen on
+    :return: instance of server
+    """
+    global _node_instance
+    if _node_instance is None:
+        raise TypeError("DHT has not been initilized")
+    return _node_instance
+
+
+def initialize_dht(
+    bootstrap_ip_port_pair_list: List[Tuple[str, int]],
+    listen_port: int,
+) -> None:
+    """
+    connects to the distributed hash table
+    if no bootstrap ip port pair list is given, it starts a new dht
+    :param bootstrap_ip_port_pair_list:
+    list of ip port tuples to connect to the dht
+    :param listen_port: port to listen on
+    :return: instance of server
+    """
+    global _node_instance
+
+    get_logger("kademlia")
```

Comment:
I have an idea what this might be doing, but could you explain why this is here?